db/write_batch.cc

```diff
@@ -1112,9 +1112,23 @@ Status WriteBatch::SingleDelete(ColumnFamilyHandle* column_family,
       this, GetColumnFamilyID(column_family), key);
 }
 
-Status WriteBatchInternal::DeleteRange(WriteBatch* b, uint32_t column_family_id,
+Status WriteBatchInternal::DeleteRange(WriteBatch* b,
+                                       uint32_t column_family_id,
+                                       const Slice& begin_key,
+                                       const Slice& end_key) {
+  return WriteBatchInternal::DeleteRange(b, nullptr, column_family_id, begin_key, end_key);
+}
+
+Status WriteBatchInternal::DeleteRange(WriteBatch* b,
+                                       ColumnFamilyHandle* handle,
+                                       uint32_t column_family_id,
```

Comment:
Rather than taking in both a handle and an ID, can you make a public API that takes in the handle and a private one that takes the ID and have the public one call the private?  Or are there still place you will not have a handle?

Reply:
If not, can you please assert that the input handle has the same ID?

Reply:
@mrambacher One question, what's the relation between column family handle and column family id? Could we reconstruct CF handle based off CF id?